core/consensus/babe/primitives/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Primitives for BABE.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use runtime_primitives::ConsensusEngineId;
+use substrate_client::decl_runtime_apis;
+
+/// The `ConsensusEngineId` of BABE.
+pub const BABE_ENGINE_ID: ConsensusEngineId = [b'b', b'a', b'b', b'e'];
+
+decl_runtime_apis! {
+    /// API necessary for block authorship with BABE.
+    pub trait BabeApi {
+        /// Return the slot duration in milliseconds for BABE.  Currently, only
+		/// the value provided by this type at genesis will be used.
+        ///
+        /// Dynamic slot duration may be supported in the future.
+        fn slot_duration() -> u64;
+
+        /// Return the expected block time in milliseconds for BABE.  Currently,
```

Comment:
seems like tabs and spaces are mixed in this file.

Reply:
@rphmeier can we add a `rustfmt.toml`?  Then we can just use `rustfmt` and everything will Just Work.

Reply:
unfortunately, not -- it doesn't quite encode the style that we use

Reply:
Does nightly `rustfmt` have enough options?  Could we file PRs to add the features we need?